### DISCUSSION TITLE
making header-only library package in vcpkg?

### DISCUSSION THREAD

**Discussion by mbandu** [microsoft/vcpkg#37604] [Q&A] [answered]
Hi,

How can I port header-only library package (cmake) with vcpkg? 
When i try, i cannot get my .hpp included in the package. 
please advise? thank you so much.

**Comment by Osyotr** [comment, marked as answer]
If your library has proper install steps, see https://github.com/microsoft/vcpkg/tree/master/ports/ms-gsl
If your library doesn't have install steps, see https://github.com/microsoft/vcpkg/tree/master/ports/earcut-hpp

**Reply by mbandu** [reply]
@Osyotr thank you so much.

**Reply by Zitrax** [reply]
How does it work in general, say for example the boost libraries. Is there any way I can add them in vcpkg.json and have them used as header-only, i.e. not having them compiled at all? (I know I can have them built and still use them in a header-only way - but I am specifically wondering if the built part can be disabled if I am not going to use it anyway)

Or is it something that have to be supported per package level for example with features?

**Reply by Osyotr** [reply]
You can skip building of a library in an overlay-port but IMHO such setup is fragile and I wouldn't use it. Boost libraries don't take too long to build anyways.

**Reply by Zitrax** [reply]
I had to use overlays in the end since I did not find a way to configure the boost packages to build without thread support (the target platform lacks it).
